### ISSUE TITLE
Signatures of invokable objects should be checked

### ISSUE THREAD

**Issue by weirdan** [vimeo/psalm#626]
```php
<?php
  class A {
  	public function __invoke(string $p): void {}
  }
  
  $q = new A;
  $q(1); // should complain
  (new A)(1); // should complain
  (new A)->__invoke(1); // does complain, and so should previous two
```
